traffic_ops/traffic_ops_golang/topology/topologies.go

```diff
@@ -150,6 +150,19 @@ func (topology *TOTopology) Validate() error {
 	rules["empty cachegroups"] = topology_validation.CheckForEmptyCacheGroups(topology.ReqInfo.Tx, cacheGroupIds, dsCDNs, false, nil)
 	rules["required capabilities"] = topology.validateDSRequiredCapabilities()
 
+	for _, cdnId := range dsCDNs {
+		err, dsId := dbhelpers.GetDSByCDNIdTopology(topology.ReqInfo.Tx.Tx, cdnId, topology.Name)
+		if err != nil {
+			return fmt.Errorf("failed to get a response from DB: %s", err)
+		}
+		//Get current Topology-CG for the requested change.
+		topoCachegroupNames := topology.getCachegroupNames()
+		userErr, sysErr, _ := dbhelpers.CheckTopologyOrgServerCGInDSCG(topology.ReqInfo.Tx.Tx, dsId, topology.Name, topoCachegroupNames)
+		if userErr != nil || sysErr != nil {
+			return fmt.Errorf("%s", userErr.Error())
```

Comment:
If `userErr != nil`, we should just `return userErr`. But if `sysErr != nil`, we should log the error then `return errors.New("unable to validate topology")`

Reply:
Done.